IMP-/Rel_pow.thy

```diff
@@ -0,0 +1,37 @@
+ (*
+    Authors: Bilel Ghorbel, Florian Kessler
+ *)
+section "Inductive definition of relation power"
+theory Rel_pow imports Main begin
+
+inductive rel_pow:: "('a \<Rightarrow> 'a \<Rightarrow> bool) \<Rightarrow> nat \<Rightarrow> 'a \<Rightarrow> 'a \<Rightarrow> bool" for r where
+refl: "rel_pow r 0 x x"|
+step:"r x y \<Longrightarrow> rel_pow r n y z \<Longrightarrow> rel_pow r (Suc n) x z"
+hide_fact (open) refl step
```

Comment:
can you explain why you defined it yourself instead of using what is there?

Reply:
If you're meaning the relation power as defined `HOL.Transitive_Closure.` 
Then it's because (and I believe this is also the reason why the IMP theory defined its own Star theory instead of using the definition in HOL) that Isabelle's value command wouldn't be able to compute the predicates involving `small_step_pow`. It requires code to be generated. A thing that seems not to be supported in HOL definition, since it is not inductive.

I believe this is a really useful feature, not only would've that helped us through debugging, but I guess it would help us while working on the next issues to be able to compute some examples.

Reply:
okay. if you like you can add a short comment to that theory explaining exactly that.